### PR TITLE
Remove StartupProbe featuregate and related logic

### PR DESCRIPTION
**Pull request by matthyx**
**What type of PR is this?**

/kind cleanup
/sig node

**What this PR does / why we need it**:

This PR removes `StartupProbe` featuregate and related logic, to promote this feature to GA in 1.20.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/950
- [Usage]: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes
```

### CODE REVIEW

**Review by thockin — COMMENTED** [review]
API changes look good.  Glad to see this come to a conclusion.

Have we had any sort of signals/fedback about this feature?

File: test/e2e_node/startup_probe_test.go [hunk 1]
```diff
@@ -1,219 +0,0 @@
...
-	imageutils "k8s.io/kubernetes/test/utils/image"
-
-	"github.com/onsi/ginkgo"
-)
-
-const (
-	defaultObservationTimeout = time.Minute * 4
-)
-
-var _ = framework.KubeDescribe("StartupProbe [Serial] [Disruptive]", func() {
```
[the remark this answers is not in the document]
Comment (SergeyKanzhelev) [reply test/e2e_node/startup_probe_test.go:41]: https://github.com/kubernetes/test-infra/pull/18971

It turns out there is more into this tab than just removing the feature flag. But feature flag can be deleted right away anyway.

**Review by thockin — COMMENTED** [review]
/approve

**Comment by MHBauer** [comment]
Awesome. This gets it out of the miscellaneous uncategorized orphans test job!

**Pull request merged**